### DISCUSSION TITLE
Proxy Manager and AccessToken.expiresIn

### DISCUSSION THREAD

**Discussion by SalvadorCali** [mia-platform/community#10] [Q&A] [answered]
Hi everyone,
working on the addition of `http-proxy-manager` on a project, we noticed that the access token received directly from Microsoft contains the field `expires_in` as string although the standard and their documentation specifies that it should be a number:
https://www.rfc-editor.org/rfc/rfc6749#section-5.1
https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow

The `http-proxy-manager` doesn't support the string format for the `expires_in` field and we should conform to the standard.

The first idea was to extend the `http-proxy-manager` to handle also string values, but in this way we'll modify the service only to manage this corner case. 

How can we handle this situation?

**Comment by davidebianchi** [comment, marked as answer]
Hi @SalvadorCali, can you share the configuration of the http-proxy-manager? Without the credentials.

**Reply by SalvadorCali** [reply]
Yes, sure!
This is the config file:
```
{
  "proxies": [
    {
      "authentication": "oauth2",
      "clientId": "{{DYNAMICS_CLIENT_ID}}",
      "clientSecret": "{{DYNAMICS_CLIENT_SECRET}}",
      "tokenIssuerUrl": "{{DYNAMICS_AUTHORITY_HOST_URL}}/{{DYNAMICS_TENANT}}/oauth2/token",
      "targetBaseUrl": "{{CONTACT_CENTER_BASE_URL}}",
      "basePath": "/basePath1",
      "grantType": "client_credentials",
      "authType": "client_secret_basic",
      "additionalAuthFields": {
        "resource": "{{DYNAMICS_RESOURCE}}"
      }
    },
    {
      "authentication": "oauth2",
      "clientId": "{{DYNAMICS_CLIENT_ID}}",
      "clientSecret": "{{DYNAMICS_CLIENT_SECRET}}",
      "tokenIssuerUrl": "{{DYNAMICS_AUTHORITY_HOST_URL}}/{{DYNAMICS_TENANT}}/oauth2/token",
      "targetBaseUrl": "{{CONTACT_CENTER_BASE_URL}}",
      "basePath": "/baseParh2",
      "grantType": "client_credentials",
      "authType": "client_secret_basic",
      "additionalAuthFields": {
        "resource": "{{DYNAMICS_RESOURCE}}"
      }
    }
  ]
}
```
And these are the environment variables:
![image](https://user-images.githubusercontent.com/37811007/189896571-d74240c3-4f51-44b6-a944-b2b9010fba3f.png)

Anyway, due to the semplicity of my proposed solution, I opened a merge request on the repository.

**Reply by davidebianchi** [reply]
Are you using the azure active directory B2C? The standard active directory seems to return correctly a number.

Here some links, the issue still seems to be so:
https://docs.microsoft.com/en-us/answers/questions/877384/azure-active-directory-b2c-authorization-code-flow.html
https://feedback.azure.com/d365community/idea/7772fd95-26e6-ec11-a81b-0022484ee92d

Probably, it's common interest to enhance the service to also include this specific corner side to support also azure active directory B2C.
